File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        b = s
        max_length = 0
        
        for i in range(len(b)):
            temp = ""
            for j in range(i, len(b)):
                if b[j] not in temp:
                    temp += b[j]
                else:
                    break
            
            max_length = max(max_length, len(temp))
        
        return max_length

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int maxLen = 0;

        for(int i = 0; i < s.length(); i++) {
            String temp = "";

            for(int j = i; j < s.length(); j++) {
                if(temp.indexOf(s.charAt(j)) != -1) {
                    break;
                }
                temp += s.charAt(j);
            }

            maxLen = Math.max(maxLen, temp.length());
        }

        return maxLen;
    }
}